### PR TITLE
plat-stm32mp1: move platform drivers to clk framework

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/drivers/stm32_uart.c [hunk 1]
```diff
@@ -149,7 +152,19 @@ struct stm32_uart_pdata *stm32_uart_init_from_dt_node(void *fdt, int node)
 	pd->chip.ops = &stm32_uart_serial_ops;
 	pd->base.pa = info.reg;
 	pd->secure = (info.status == DT_STATUS_OK_SEC);
-	pd->clock = (unsigned int)info.clock;
+
+	if (fdt_getprop(fdt, node, "clocks", NULL)) {
```
Comment (ldebieve) [inline core/drivers/stm32_uart.c:156]: Clocks is part of the mandatory properties listed in the kernel Documentation/devicetree/bindings/serial/st,stm32-uart.yaml.
We do not expect the uart to not have clocks property here.

File: core/drivers/stm32_uart.c [hunk 2]
```diff
@@ -149,7 +152,19 @@ struct stm32_uart_pdata *stm32_uart_init_from_dt_node(void *fdt, int node)
...
+		res = clk_dt_get_by_index(fdt, node, 0, &pd->clock);
+		if (!res) {
+			assert(pd->clock);
+
+			res = clk_enable(pd->clock);
+			if (res)
+				panic();
+		}
+	}
+	if (!pd->clock)
```
Comment (ldebieve) [inline core/drivers/stm32_uart.c:166]: missing empty line before

File: core/drivers/stm32_uart.c (hunk 1 above)
Comment (etienne-lms) [reply core/drivers/stm32_uart.c:156]: Ok, i'll remove the test.

File: core/drivers/stm32_uart.c (hunk 2 above)
Comment (etienne-lms) [reply core/drivers/stm32_uart.c:166]: I'll discard this change since your comment above.

**Comment by ldebieve** [comment]
`Reviewed-by: Lionel Debieve <lionel.debieve@foss.st.com>`

**Comment by etienne-lms** [comment]
fixup commit squashed and tag applied.

**Comment by jenswi-linaro** [comment]
This looks safe to merge since it only touches on stm32 stuff.

**Comment by jforissier** [comment]
> This looks safe to merge since it only touches on stm32 stuff.

Yes, and @etienne-lms has [confirmed](https://github.com/OP-TEE/optee_os/pull/5094#issuecomment-1023113123) he has tested with this too. Thanks!

**Pull request merged**